actionview/test/template/render_test.rb

```diff
@@ -312,11 +312,11 @@ def test_render_renderable_with_incompatible_object
     end
   end
 
-  def test_render_renderable_does_not_mask_nomethoderror_from_within_render_in
+  def test_render_renderable_does_not_mask_nameerror_from_within_render_in
     renderable = Object.new
-    renderable.define_singleton_method(:render_in) { |*| nil.foo }
+    renderable.define_singleton_method(:render_in) { |*| nil.method(:foo) }
 
-    assert_raises NoMethodError, match: "undefined method `foo' for nil" do
+    assert_raises NameError, match: "undefined method `foo' for class `NilClass'" do
```

Comment:
I don't think this should change.  Presumably, we will remove the arity check when we remove the deprecated code, and go back to rescuing `NoMethodError`.  If we forget to also change this test back, we will lose test coverage.

Conversely, raising `NoMethodError` in this test can exercise both `rescue NameError` and `rescue NoMethodError`.